packages/overlay/src/vaadin-overlay-focus-mixin.js

```diff
@@ -111,7 +111,11 @@ export const OverlayFocusMixin = (superClass) =>
      */
     _trapFocus() {
       if (this.focusTrap) {
-        this.__focusTrapController.trapFocus(this._focusTrapRoot);
+        const root = this._focusTrapRoot;
+        // Ensure the focus trap root element is not hidden
+        if (getComputedStyle(root).display !== 'none') {
```

Comment:
This solution only stops focus from being trapped while the root element is hidden but doesn't re-enable it once the root is visible again.